BinaryTreeTest.java

```diff
@@ -0,0 +1,67 @@
+package com.katermar.assignment2;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Created by USER on 5/31/2017.
+ */
+public class BinaryTreeTest {
+
+    public static void main(String args[])
+    {
+        BinaryTree tree = new BinaryTree();
+
+        /* Construct the following binary tree
+                  2
+                /   \
+               1     4
+                    /  \
+                   3    5
+                         \
+                          7
+        */
+        int[] array = {2, 1, 4, 3, 5, 7};
+        tree.populateTree(array);
+        System.out.println();
+
+        List<BinaryTree.TreeNode> ancestorList = new LinkedList<>();
+        if (!tree.findAllAncestors(ancestorList, tree.root, 7)) {
+            System.out.println("Element doesn't exist");
+        } else {
+            System.out.println("Ancestors - " + ancestorList);
+        }
+
+        System.out.println();
+        checkLCA(4, tree, 7, 3);
+        checkLCA(2, tree, 2, 1);
+        checkLCA(10, tree, 2, 1);
+    }
+
+    public static String findLCAtest(BinaryTree tree, int a, int b) {

```

Comment:
It is not really needed any more - the whole check happens in checkLCA. Of course you can keep this method if you wish, but please remove call to it in checkLCA.